src/node/internal/internal_stringdecoder.ts

```diff
@@ -43,15 +43,15 @@ const kBufferedBytes = 5;
 const kEncoding = 6;
 const kSize = 7;
 
-const encodings : Record<string,number> = {
-  ascii: 0,
-  latin1: 1,
-  utf8: 2,
-  utf16le: 3,
-  base64: 4,
-  base64url: 5,
-  hex: 6,
-};
+const encodings = [
+  'ascii',
+  'latin1',
+  'utf8',
+  'utf16le',
+  'base64',
+  'base64url',
+  'hex',
+];
```

Comment:
This array needs to be in the exact same order as the C++ enum or it breaks. I would add a comment about this, if there isn't a better way to do it.

Reply:
Added a comment.